lib/Controller/SettingsController.php

```diff
@@ -68,9 +68,17 @@ public function enable($state) {
 		$user = $this->userSession->getUser();
 		if ($state) {
 			$secret = $this->totp->createSecret($user);
-			
 			$qrCode = new QrCode();
-			$qr = $qrCode->setText("otpauth://totp/ownCloud%20TOTP?secret=$secret")
+			$nameLine = $user->getDisplayName();
+			$loginName = \OC::$server->getUserSession()->getSession()->get('loginname');
```

Comment:
is 'loginname' always set? This looks error-prone, an app developer wouldn't know that. Maybe we should think about a proper public API method to get the login name?